lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -426,6 +426,60 @@ database, or delete the offending rows through PgAdmin III or psql'),
   max_version => '1.4'
 );
 
+push @tests, __PACKAGE__->new(
+   test_query => 'select *, eca.id as id  from entity_credit_account eca
+                     join entity_class ec on eca.entity_class = ec.id
+                     join entity e on eca.entity_id = e.id
+                   where meta_number in
+                       (select meta_number from entity_credit_account
+                        group by meta_number having count(*) > 1)
+                   order by meta_number',
+ display_name => marktext('No duplicate meta_numbers'),
+         name => 'no_meta_number_dupes',
+ display_cols => [ 'meta_number', 'class', 'description', 'name' ],
+      columns => ['meta_number'],
+        table => 'entity_credit_account',
+ instructions => marktext('Make sure all meta numbers are unique.'),
+      appname => 'ledgersmb',
+  min_version => '1.3',
+  max_version => '1.4'
+);
+
+#=pod
+
+# This must be done before the acc_trans test, for they have duplicates
+# and need a unique key
```

Comment:
actually, this will *always* run before the acc_trans test, even if the acc_trans test's UI has already been shown and "corrective information" has been provided.

There's one problem here though: because SQL doesn't guarantee set-ordering, the code below doesn't necessarily generate the same unique values for the same rows as the ones already in the "uniquified" set (and thus in the corrective data).